### PR TITLE
Updated Organization param from location to Address Type.

### PR DESCRIPTION
**Pull request by Anubhav-2003**
**What kind of change does this PR introduce?**

A feature

**Issue Number:**
Fixes:  #1650 

**Did you add tests for your changes?**
Yes

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**
Yes

The PR changes the location field in Organizations to AddressInput Type. All front-end applications using the API must change their queries, mutations, forms, components to accommodate for the change. 
**Other information**

The PR is still in work. Creating PR for review, and suggestions.
**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by Anubhav-2003** [comment]
@palisadoes Respected sir, I have created this PR for your review, and also update with my status. It is still under work. I have tried to make changes from location to address in Organizations in all the places I could figure out. Yet, I may be missing some places. Also, I have reduced the sample data for testing purposes,  adhering to the new structure of the organization type, but it is only temporary and I will put all the existing organizations with the new address field before merge. 

I have also updated and written some new test cases for the createOrganiztions.ts file, I will be writing more finally after getting your feedback on the actual file changes. Also I tried to make some changes in the Talawa-admin repo, but found out that there are a lot of files where this change is needed, before I can actually verify if the functionality is working properly. 

So someone will have to take the initiative of making all the changes in the Talawa-admin, which I think are plenty in number. Because unless the front-end sends the correct requests to the API, I am not sure about the changes. 

Thank you.

**Comment by palisadoes** [comment]
Please fix the failing tests and let us know when you are ready for a review

**Comment by Anubhav-2003** [comment]
@palisadoes Sir, I have fixed the failing test cases. Can my code be please reviewed and given some insight as to if I need to make some more changes.

Thank you.

**Comment by palisadoes** [comment]
Reviewers have been assigned
Please fix the conflicting files

File: sample_data/organizations.json
Comment (xoldyckk) [inline sample_data/organizations.json:23]: why were some smaple organizations deleted here?

File: src/typeDefs/inputs.ts
Comment (xoldyckk) [inline src/typeDefs/inputs.ts:147]: @palisadoes can organizations choose to give no address information?

Comment (palisadoes) [reply src/typeDefs/inputs.ts:147]: Make it mandatory

File: sample_data/organizations.json
Comment (Anubhav-2003) [reply sample_data/organizations.json:23]: Actually Sir, I removed the sample organisations, and just kept one updating its located param to address just for testing purposes. I will bring back all the existing organisations before the PR merges.

File: src/typeDefs/inputs.ts
Comment (Anubhav-2003) [reply src/typeDefs/inputs.ts:147]: @palisadoes ok sir. I will change it to mandatory.

**Comment by Anubhav-2003** [comment]
@palisadoes Respected sir, I have fixed all the merge conflicts. Actually the conflict was due to this PR #1643 which was merged yesterday, which added a new field 'userRegistrationRequired' to organization type and removed 'isPublic' field. But my PR had these fields. So I had to change my sections of the code accordingly. 

Thank you.

File: src/typeDefs/inputs.ts
Comment (Anubhav-2003) [reply src/typeDefs/inputs.ts:141]: @palisadoes @xoldyckk Sir, I have  made the address field mandatory for creation of organizations. Thank You.

Comment (xoldyckk) [reply src/typeDefs/inputs.ts:141]: @palisadoes which of the address fields [here](https://github.com/PalisadoesFoundation/talawa-api/blob/46771dbdb3013f9a548a5b331cd2247e26032462/src/typeDefs/types.ts#L361) should be mandatory?

**Comment by palisadoes** [comment]
Reviewers. This should be complete. Please take a look
- @xoldyckk
- @JamaicanFriedChicken
- @sumitra19jha
- @DMills27

File: src/typeDefs/inputs.ts
Comment (xoldyckk) [inline src/typeDefs/inputs.ts:141]: @Anubhav-2003 make it non nullable in the input

Comment (palisadoes) [reply src/typeDefs/inputs.ts:141]: None to make it easier for the admin, but `Country Code` should come from a list of enums when updated

**Comment by palisadoes** [comment]
Please fix the conflicting files

File: src/typeDefs/inputs.ts
Comment (Anubhav-2003) [reply src/typeDefs/inputs.ts]: @xoldyckk Sir, I have made it non-nullable in my last commit.

Comment (Anubhav-2003) [reply src/typeDefs/inputs.ts]: @palisadoes  @xoldyckk Sir, I think in the front-end we should use an API to get the country code, and then use enums to verify it in the backend. There has to be major changes in the Talawa Admin, from updating queries, mutations, to using API's to send the data to the backend before this PR can be merged. Otherwise the Talawa-Admin portal would break.

**Comment by Anubhav-2003** [comment]
@palisadoes @xoldyckk Respected sirs, I have resolved all the conflicts as well as brought back all the sample organizations, updating their structure to have address param instead of location.

**Comment by Anubhav-2003** [comment]
@palisadoes Any update on my PR sir. Also shall I start working on talawa admin to administer the changes required to work with these changes in the API ?

**Comment by palisadoes** [comment]
This really needs attention. Please review.

1. @JamaicanFriedChicken
1. @sumitra19jha
1. @DMills27
1. @xoldyckk

File: sample_data/organizations.json
Comment (Anubhav-2003) [reply sample_data/organizations.json:61]: @sumitra19jha Actually I did not make that change, while I was working in the issue some other PR got merged into develop branch, that had made the change. I just resolved the merge conflicts, keeping the changes in the updated develop. Even I was confused as to why it was made. My only change involves replacing location param with Address. 

Please let me know if I can help with anything else.

Thank You.

**Comment by Anubhav-2003** [comment]
@palisadoes Respected sir, Can I start working on the frontend implementation in Talawa-admin for these changes in the API. Unless those are made, this PR can not be merged. Otherwise the admin portal would break.

Thank You.

**Comment by sumitra19jha** [comment]
Good to merge

**Review by palisadoes — APPROVED** [review]

**Comment by Anubhav-2003** [comment]
@palisadoes Respected sir, please don't merge it. Let me make the changes in the front end talawa admin first. Or else it will break. 

Thank you.

**Comment by palisadoes** [comment]
When will you submit the Admin PR? Getting reviewers for resubmitting this PR once again is going to add delays

**Comment by Anubhav-2003** [comment]
> When will you submit the Admin PR? Getting reviewers for resubmitting this PR once again is going to add delays

Sir, as proposed by me earlier in the issue #1650, you approved for the issues to be solved simultaneously after this PR got approved before merging. But sir, there are multiple places in the frontend where changes are required which must be tackled by the issue https://github.com/PalisadoesFoundation/talawa-admin/issues/1398, like updating the UI, mutations, queries, and most importantly calling an API, to fetch the actual address data in the frontend to send the required fields. I can make those changes, but i am not assigned to this issue. If you authorise me i can make those changes. For now the API server work is completely done, but unless the frontend changes are made the API will throw an error. 

Thank you.

**Comment by palisadoes** [comment]
OK. I'll let @Anvita0305 know to base the changes in that PR on this one

**Comment by Anubhav-2003** [comment]
@palisadoes ok sir, but if there are any delays on that issue, I will be more than obliged to make the changes myself in the admin portal, by creating an issue and asking for assignment. I will try to make the changes as early as possible. Please let me know if you need me to do that.

Thank you.

**Comment by palisadoes** [comment]
@Anubhav-2003 you can ask to be assigned this issue
- https://github.com/PalisadoesFoundation/talawa-admin/issues/1457

**Pull request merged**
